app/main/views/buyers.py

```diff
@@ -436,7 +436,7 @@ def award_brief_details(framework_slug, lot_slug, brief_id, brief_response_id):
     if not is_brief_correct(brief, framework_slug, lot_slug, current_user.id):
         abort(404)
     brief_response = data_api_client.get_brief_response(brief_response_id)["briefResponses"]
-    if not brief_response.get('status') == 'pending-awarded':
+    if not brief_response.get('status') == 'pending-awarded' or not brief_response.get('briefId') == brief.get('id'):
```

Comment:
Slight preference for `A != B or C != D` being more readable than `not A == B or not C == D`, but don't feel obliged to change it.